### ISSUE TITLE
posix activate script can't identify dash without ps

### ISSUE THREAD

**Issue by johnflavin** [conda/conda-pack#170]
I am trying to use conda-pack to build a docker image that packages a conda environment without having conda in the final image. I'm essentially following this guide: [Shrink your Conda Docker images with conda-pack](https://pythonspeed.com/articles/conda-docker-image-size/).

The issue I am having is that my base image (the final one, not the build image) only has `/bin/sh` and not `/bin/bash` or some other shell. And the conda-pack activation script does not seem to work in `/bin/sh`. When I try to activate the environment within the image I get this error:

```shell
$ . /env/bin/activate
Unrecognized shell.
```

I see there is #72 to support other shells, but that issue says that "posix" shell is supported. That doesn't match my experience. Is that issue incorrect? Am I doing something wrong?

**Comment by xhochy** [comment]
What is the backing implementation of `/bin/sh` in your case? Can you e.g. post the output of `ps -p$$ -o cmd="",comm="",fname="" 2>/dev/null | sed 's/^-//' | grep -oE '\w+' | head -n1` as used in the script?
https://github.com/conda/conda-pack/blob/7323db1bcc8016a3f7aea54c0d44c860de7387b3/conda_pack/scripts/posix/activate#L13

**Comment by johnflavin** [comment]
There is no output from that pipeline. But if I remove all the downstream pipelines and output redirection from the first command, I do see the source of the problem: `ps` is not installed.
```shell
$ ps -p$$ -o cmd="",comm="",fname=""
/bin/sh: 20: ps: not found
```

**Comment by xhochy** [comment]
Then you need to find another way to tell us which shell you are using. Without a way to detect the shell, we probably cannot support your use case.

**Comment by johnflavin** [comment]
In the activate script you linked above, that `ps` command is beneath a comment linking to a stack overflow post where that command seemingly came from: https://unix.stackexchange.com/a/120138/92065. I looked around on that page and found a different command that did work for me.
```shell
$ basename $(readlink /proc/$$/exe)
dash
```

I may be able to work around this on my end by swapping the `ps` pipeline with this `readlink` command.

**Comment by johnflavin** [comment]
I have been able to work around the my issues by using `sed` to remove lines in the activate script that run commands I don't have and replace those lines with what the output should be. (Because not only does my image not have `ps` it also does not have `lsof`, which caused another error later in the script.)

I don't know if you will want to close the issue since my use case has been solved or leave it open because other people may run into this.

Either way, thanks for your help!
